### PR TITLE
Update Wifibroadcast HOWTO

### PR DESCRIPTION
**Pull request by svpcom**
I've updated Wifibroadcast HOWTO due to old wifi hardware EOL and significant changes in code and config files.

### CODE REVIEW

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 1]
```diff
@@ -1,23 +1,26 @@
-# Long-distance Video Streaming in QGroundControl
+# Long-distance Video Streaming and telemetry via raw WiFi radio
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:1]: First level capitalisation: Long-distance Video Streaming and Telemetry via Raw WiFi radio

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
 
-This page shows how to set up a companion computer with a camera (Logitech C920 or RaspberryPi camera) such that the video stream is transferred from the UAV to a ground computer and displayed in *QGroundControl*. This setup uses WiFi in unconnected (broadcast) mode and software from the [Wifibroadcast project](https://github.com/svpcom/wifibroadcast/wiki).
+This page shows how to set up a companion computer with a camera (Logitech C920 or RaspberryPi camera) such that the video stream is transferred from the UAV to a ground computer and displayed in *QGroundControl*. Also it provides support for bidirectional telemetry like SiK radio does. This setup uses WiFi in unconnected (broadcast) mode and software from the [Wifibroadcast project](https://github.com/svpcom/wifibroadcast/wiki).
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:3]: Perhaps "Also it provides support for bidirectional telemetry like SiK radio does. " TO "The mechanism also provide a bidirectional telemetry link (i.e. like SiK radio). "

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 
 
 ## Wifibroadcast Overview
 
-The *Wifibroadcast project* aims to mimic the advantageous properties of using an analog link to transmit HD video (and other) data when using WiFi radios. For example, it attempts to provide a video feed that degrades gracefully with signal degradation/distance.
+The *Wifibroadcast project* provides video and telemetry transport which use low-level WiFi packets to avoid distance and latency limitations of ordinary ieee80211 stack.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:8]: ieee80211 should be: IEEE 802.11
If you can provide a link to the standard that would be extra cool.

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
 
-- Minimal latency by encoding every incoming RTP packet to a single WiFi (IEEE80211) packet and immediately sending (doesn't serialize to byte stream).
-- Smart FEC support (immediately yield packet to video decoder if FEC pipeline without gaps).
-- Stream encryption and authentication ([libsodium](https://download.libsodium.org/doc/))
-- Distributed operation. It can gather data from cards on different hosts, so that bandwidth is not limited to that of a single USB bus.
-- Aggregation of MAVLink packets. It doesn't send WiFi packet for every MAVLink packet.
-- [Enhanced OSD for Raspberry Pi](https://github.com/svpcom/wifibroadcast_osd) (consumes 10% CPU on Pi Zero).
-
+ - 1:1 map RTP to IEEE80211 packets for minimum latency (doesn't serialize to byte steam)
+ - Smart FEC support (immediately yeild packet to video decoder if FEC pipeline without gaps)
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:15]: typo yeild > should be yield.

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 5]
```diff
@@ -1,23 +1,26 @@
...
+ - [Bidirectional mavlink telemetry](https://github.com/svpcom/wifibroadcast/wiki/Setup-HOWTO). You can use it for mavlink up/down and video down link.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:16]: everywhere "mavlink" should be MAVLink

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 6]
```diff
@@ -1,23 +1,26 @@
...
 The high level benefits of *Wifibroadcast* include:
 
-- Minimal latency by encoding every incoming RTP packet to a single WiFi (IEEE80211) packet and immediately sending (doesn't serialize to byte stream).
-- Smart FEC support (immediately yield packet to video decoder if FEC pipeline without gaps).
-- Stream encryption and authentication ([libsodium](https://download.libsodium.org/doc/))
-- Distributed operation. It can gather data from cards on different hosts, so that bandwidth is not limited to that of a single USB bus.
-- Aggregation of MAVLink packets. It doesn't send WiFi packet for every MAVLink packet.
-- [Enhanced OSD for Raspberry Pi](https://github.com/svpcom/wifibroadcast_osd) (consumes 10% CPU on Pi Zero).
-
+ - 1:1 map RTP to IEEE80211 packets for minimum latency (doesn't serialize to byte steam)
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:14]: As above, IEEE 802.11

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 7]
```diff
@@ -1,23 +1,26 @@
...
+ - Smart FEC support (immediately yeild packet to video decoder if FEC pipeline without gaps)
+ - [Bidirectional mavlink telemetry](https://github.com/svpcom/wifibroadcast/wiki/Setup-HOWTO). You can use it for mavlink up/down and video down link.
+ - Automatic TX diversity (select TX card based on RX RSSI)
+ - Stream encryption and authentication ([libsodium](https://download.libsodium.org/doc/))
+ - Distributed operation. It can gather data from cards on different hosts. So you don't limited to bandwidth of single USB bus.
+ - Aggreagation of mavlink packets. Doesn't send wifi packet for every mavlink packet.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:20]: Typo Aggreagation > Aggregation

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 8]
```diff
@@ -47,123 +51,71 @@ So you need to connect it to 5V BEC directly. You can do this two ways:
...
-
+## Tuning radio settings
+With default settings WFB use radio channel 165 (5825 MHz), width 20MHz, MCS #1 (QPSK 1/2) with long GI. This provides ~7 mbit/s of **effective** speed (i.e. usable speed after FEC and packet encoding) for **both directions** in sum, because WiFi is half-duplex. So it is sutable for video down stream 720p@49fps (4 mbit/s) + two full-speed telemetry streams (uplink and downlink). If you need a higher bandwidth you can use other MCS index (for example 2 or greater) and/or 40MHz channel. 
 
+## Antennas and diversity
+For simple cases you can use omnidirectional antennas with linear (that bundled with wifi cards) or circular (cleaver leaf) polarization. If you want to setup long distance link you can use multipe wifi adapters with directional and omnidirectional antennas. TX/RX diversity for multiple adapters supported out of box (just add multiple NICs to ``/etc/default/wifibroadcast``). If your WiFi adapter has two antennas (like Alfa AWU036ACH) TX diversity is implemented via [STBC](https://en.wikipedia.org/wiki/Space%E2%80%93time_block_code). Cards with 4 ports (like Alfa AWUS1900) are currently not supported for TX diversity (only RX is supported).
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:94]: Typo: multipe > multiple

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 9]
```diff
@@ -47,123 +51,71 @@ So you need to connect it to 5V BEC directly. You can do this two ways:
...
 
 ## FAQ
+Q: What is a difference from original wifibroadcast?
 
+A: Original version of wifibroadcast use a byte-stream as input and splits it to packets of fixed size (1024 by default). If radio packet was lost and this is not corrected by FEC you'll got a hole at random (unexpected) place of stream. This is especially bad if data protocol is not resistent to (was not desired for) such random erasures. So i've rewrite it to use UDP as data source and pack one source UDP packet into one radio packet. Radio packets now have variable size depends on payload size. This is reduces a video latency a lot.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:99]: Typo resistent > resistant

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 10]
```diff
@@ -47,123 +51,71 @@ So you need to connect it to 5V BEC directly. You can do this two ways:
...
-The new version has been rewritten to use UDP as data source and pack one source UDP packet into one radio packet. Radio packets now have variable size depends on payload size. This is reduces a video latency a lot.
-
-#### What type of data can be transmitted using Wifibroadcast?
+Q: What type of data can be transmitted using wifibroadcast?
 
-Any UDP with packet size <= 1466. For example x264 inside RTP or MAVLink.
+A: Any UDP with packet size <= 1466. For example x264 inside RTP or Mavlink.
 
-#### What are transmission guarantees?
+Q: What are transmission guarancies?
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:105]: typo: guarancies > guarantees

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 11]
```diff
@@ -47,123 +51,71 @@ So you need to connect it to 5V BEC directly. You can do this two ways:
...
 
-*Wifibroadcast* uses Forward Error Correction (FEC) which can recover 4 lost packets from a 12 packet block with default settings.
-You can tune both TX and RX (simultaneously) to fit your needs.
+A: Wifibrodcast use FEC (forward error correction) which can recover 4 lost packets from 12 packets block with default settings. You can tune it (both TX and RX simultaniuosly!) to fit your needs.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:107]: Typo: simultaniuosly > simultaneously

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 12]
```diff
@@ -47,123 +51,71 @@ So you need to connect it to 5V BEC directly. You can do this two ways:
...
 
-This can be due to:
-1. Signal power is too low. Use high-power card or antennas with more gain. Use directed antenna on the RX side. Use additional RX card for diversity (add wlan2, wlan3, ... to RX program)
-2. Signal power is too high. Especially if you use 30dBm TX indoors. Try to reduce TX power (for example hack CRDA database inside kernel and make
-   several regions with power limit 10dBm and 20dBm).
-3. Interference with other WiFi. Try to change WiFi channel and/or WiFi band. 
+Q: Is only Raspberry PI supported?
 
-   > **Caution** Don't use band that the RC TX operates on! Or setup RTL properly to avoid model loss.
+A: Wifibroadcast is not tied to any GPU - it operates with UDP packets. But to get RTP stream you need a video encoder (with encode raw data from camera to x264 stream). In my case RPI is only used for video encoding (becase RPI Zero is too slow to do anything else) and all other tasks (including wifibroadcast) are done by other board (NanoPI NEO2).
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:113]: Typo: becase > because

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 13]
```diff
@@ -1,23 +1,26 @@
...
-- Distributed operation. It can gather data from cards on different hosts, so that bandwidth is not limited to that of a single USB bus.
-- Aggregation of MAVLink packets. It doesn't send WiFi packet for every MAVLink packet.
-- [Enhanced OSD for Raspberry Pi](https://github.com/svpcom/wifibroadcast_osd) (consumes 10% CPU on Pi Zero).
-
+ - 1:1 map RTP to IEEE80211 packets for minimum latency (doesn't serialize to byte steam)
+ - Smart FEC support (immediately yeild packet to video decoder if FEC pipeline without gaps)
+ - [Bidirectional mavlink telemetry](https://github.com/svpcom/wifibroadcast/wiki/Setup-HOWTO). You can use it for mavlink up/down and video down link.
+ - Automatic TX diversity (select TX card based on RX RSSI)
+ - Stream encryption and authentication ([libsodium](https://download.libsodium.org/doc/))
+ - Distributed operation. It can gather data from cards on different hosts. So you don't limited to bandwidth of single USB bus.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:19]: Change "don't limited" to "aren't limited"

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 14]
```diff
@@ -47,123 +51,71 @@ So you need to connect it to 5V BEC directly. You can do this two ways:
    
 ## Software Setup
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:53]: Perhaps a line before list stating host for which software is being installed - I assume the development computer.

> To setup the (linux) development computer:

File: en/qgc/video_streaming_wifi_broadcast.md (hunk 8 above)
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:94]: " (that bundled with wifi cards)". Do you mean (as often bundled with wifi cards) ?

Comment (hamishwillee) [reply en/qgc/video_streaming_wifi_broadcast.md:94]: What is cleaver leave polarisation?

File: en/qgc/video_streaming_wifi_broadcast.md (hunk 9 above)
Comment (hamishwillee) [reply en/qgc/video_streaming_wifi_broadcast.md:99]: > So i've rewrite ...

TO 

> I've rewritten it to ...

Comment (hamishwillee) [reply en/qgc/video_streaming_wifi_broadcast.md:99]: > Radio packets now have variable size depends on payload size. This is reduces a video latency a lot.

TO 

> Radio packets now have variable size depending on payload size. This significantly reduces video latency.

**Review by hamishwillee — APPROVED** [review]
This looks great, though I have not tested it.

There are some typos and English wording issues. You can fix those or let me know you would prefer that I do it (in which case I will merge and make fixes).

File: en/qgc/video_streaming_wifi_broadcast.md (hunk 8 above)
Comment (svpcom) [reply en/qgc/video_streaming_wifi_broadcast.md:94]: I mean circularly polarized Coverleaf Antenna
http://www.antenna-theory.com/antennas/cloverleaf.php

**Comment by svpcom** [comment]
Please do changes if it doesn’t bother you

**Comment by hamishwillee** [comment]
I've merged this and done some tidy in line with comments as a post process. Thanks very much for this!

**Pull request merged**